### ISSUE TITLE
Unable to stop

### ISSUE THREAD

**Issue by F1p** [LennartHennigs/ESPTelnet#61]
Requesting TelnetServer.stop(); doesn't close an active connection, only prevents new?

**Comment by LennartHennigs** [comment]
No, the `stop()` function calls `WiFiServer.stop()` which closes the connection:

https://github.com/arduino/esp8266/blob/1eb0645dcb82e03e239e7932ad20802c769c89e0/libraries/ESP8266WiFi/src/WiFiServer.cpp#L132

**Comment by F1p** [comment]
I observed different behaviour,

Is it possible to enable/disable the telnet server?

**Comment by LennartHennigs** [comment]
This is what `stop()`does.
You can disconnect a client via `disconnectClient(bool triggerEvent = true)` as stated in the documentation

**Comment by F1p** [comment]
When calling stop() I found new connections prevented, but already established connections would remain active until terminated by the user

**Comment by LennartHennigs** [comment]
This is fixed in the current codebase. `ESPTelnetBase::stop()` now takes an optional boolean parameter:

```cpp
void stop(bool disconnectClient = true);
```

By default it disconnects any active client before stopping the server, which resolves the issue of the server being unable to stop when a client is connected. You can call `telnet.stop()` and the connected client will be cleanly disconnected first.

**Comment by F1p** [comment]
Thanks very much
